### ISSUE TITLE
Completing Horticulturalist

### ISSUE THREAD

**Issue by SCdF** [medic/medic-webapp#3582]
Things we need to do to achieve greatness:

# So it can be used locally

 - [x] #3747 Write a local mode, that doesn't do medic-os stuff (like chmoding users etc) ( @alxndrsn is working on this )

# So it can be used in MedicOS

I've started changes in the `paving_over_the_garden` repo.

- [ ] Fix MedicOS so it builds ( @browndav and @henokgetachew are working on this)
- [ ] include the three new packages: `horticulturalist`, `medic-api` and `medic-sentinel`
- [ ] Clean up the collection of things that preload dashboard and medic (check out `garden-medic-maker.diff`, `gardenize-couchdb`, `rearrange-sources`, and note that the asset downloaded from `preload-couchdb` **also** contains a dashboard so that we have one new medic and maybe dashboard
- [ ] #3618 Make sure people can use the dashboard configuration stuff: either rip config out and put in medic-webapp, or hide the markets stuff and re-build dashboard
- [ ] Work out how logging / log rotation works. End goal: logs should go in the right place, and they should rotate like everyone else's logs

Then we can build all this locally, test it, and await thunderous applause…

# Other stuff

- [ ] get `master` builds pushing (https://github.com/medic/medic-webapp/pull/3560)

# Bonus

- [ ] If you refresh the browser too quickly it looks awful because api restarts in the middle of your refresh. Solve this:
   - Maybe by killing api, *then* moving the ddoc over and then booting it back up (instead of moving the ddoc and re-booting it). Could be achieved by horticulturalist watching a separate ddoc location so it performs the switch instead of the api endpoint
   - Maybe re-implementing app-cache with service workers (app cache is deprecated so we want to do this anyway), so that if you manage to pull down the new ddoc then you also pull down and have all its attachments and don't need api to perform a successful reboot
   - Maybe having a status URL you can query in api that exposes the state its in (including slow states like migrating), suppressing the standard refresh mechanisms and using that state instead to know when to perform the reboot
   - Or some combination of the above

**Comment by SCdF** [comment]
(Just wanted to give this a github ticket id. Feel free to remix / move out / whatever as needed)

**Comment by alxndrsn** [comment]
Can we consider renaming to `horti` for `svc-*` scripts etc.?

**Comment by SCdF** [comment]
@alxndrsn I have zero opinion about it, feel free to rename it to whatever you like :-)

**Comment by SCdF** [comment]
@garethbowen I noticed you put technical debt on this. To be clear, I think this change adds real features and fixes real bugs:
 - Deployments will be a more stable prospect over bad connections
 - No more waiting for someone to deploy and test one beta so we can release another.
 - Production logs will not magically be deleted a few hours after they are produced
 - Can deploy branches to amz instances

And probably a few more things I'm forgetting

**Comment by SCdF** [comment]
I've pulled out the build generation stuff so we can merge it and get it into master: https://github.com/medic/medic-webapp/pull/3610

**Comment by SCdF** [comment]
NB: this is mostly blocked on being able to build medic-os and run it locally.

**Comment by SCdF** [comment]
Closing in favour of https://github.com/medic/medic-webapp/issues/3867 to describe the MedicOS work needed.
